Bottom: 0ffc54a8c1529228da61a08e5bcf4342c9ae0d20
Top:    206cfdf288b25da9dfeee511eee0a36701bffec3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-05 22:23:41 +0000

Patch out non-full-screen invalidates

(XXX: Probably breaks non-composited screen repaints horribly!)


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 8e3b9f4..af391dd 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -2344,6 +2344,17 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
     object3d_test_init ();
   }
 
+//  printf ("Expose event at (%i,%i): %i x %i\n", ev->area.x, ev->area.y, ev->area.width, ev->area.height);
+//  printf ("Event type %i, send_event %i\n", ev->type, ev->send_event);
+
+  /* For some reason we get horrible glitching when we don't repaint the whole event region...
+   * It also appears that we get spurious small repaints interleaved with full area re-paints,
+   * so lets try just droping the small ones.
+   * NB: Currently, all our programmatically triggered repaints cover the whole region.
+   */
+  if (ev->area.x != 0 || ev->area.y != 0)
+    return FALSE;
+
   gtk_widget_get_allocation (widget, &allocation);
 
   ghid_start_drawing (port, widget);
